haproxy/haproxy_cmd/run.go

```diff
@@ -118,31 +118,29 @@ func execAndCapture(path string, re *regexp.Regexp) (string, error) {
 
 // CheckEnvironment Verifies that all dependencies are correct
 func CheckEnvironment(dataplaneapiBin, haproxyBin string) error {
-	errors := make([]error, 2)
+	var err error
 	wg := &sync.WaitGroup{}
 	wg.Add(2)
-	ensureVersion := func(path, rx, version string, idx int) {
+	ensureVersion := func(path, rx, version string) {
+		defer wg.Done()
 		r := regexp.MustCompile(rx)
 		v, e := execAndCapture(path, r)
 		if e != nil {
-			errors[idx] = e
+			err = e
 		} else if strings.Compare(v, "1.2") < 0 {
-			errors[idx] = fmt.Errorf("%s version must be > 1.2,, but is: %s", path, v)
+			err = fmt.Errorf("%s version must be > 1.2, but is: %s", path, v)
 		}
-		wg.Done()
 	}
-	func() {
-		ensureVersion(haproxyBin, "^HA-Proxy version ([0-9]\\.[0-9]\\.[0-9])", "2.0", 0)
+	go func() {
```

Comment:
Nitpick: this can be just `go ensureVersion(...)`

Reply:
Sure, thanks :) Updated